### ISSUE TITLE
Coins tab -> Right Click -> "Spend From" doesn't behave 100% kosher with respect to CashShuffle

### ISSUE THREAD

**Issue by cculianu** [clifordsymack/Electron-Cash#48]
- UTXOs that are frozen by cashshuffle can be spent using this method (not a huge showstopper but will make the next shuffle round fail and start over)
- The UTXO list saved in the send tab's spend from is not dynamically updated (it should be!)

A possible rewrite of this subsystem might be in order as it has potential bugs even with non-cashshuffle where if the wallet is updated it never finds out if UTXOs have disappeared.

**Comment by zquestz** [comment]
For now, when CashShuffle is enabled why don't we just disable this option in the pulldown menu? Seems like a quick fix that would resolve this for now.

**Comment by cculianu** [comment]
Hmm.  I use "Spend From" all the time, even when CashShuffle is enabled.

**Comment by zquestz** [comment]
Another option is to only enable it for shuffled utxos, then there is no contention as the shuffler won't be touching them.

**Comment by cculianu** [comment]
That's one quick fix, sure.  The other is to actually apply the rules set in the "Send" tab and pop out the coins from the list for coins that don't fit.  So if you have the "send" tab set to "spend shuffled only", it only will allow those coins to appear, for example.  

A dialog popup should also warn you that it edited your list and removed coins not compatible with the current settings in the "Send" tab.

This way users using this advanced feature are at least  told their list was edited and are made aware of the implications of spending those coins

**Comment by imaginaryusername** [comment]
Not sure which commit fixed this, but I believe the send tab now catches any coins that does not fit the shuffled/unshuffled status, and will remove them from the final tx. Good enough to close for me. =)

**Comment by cculianu** [comment]
Yeah I got obsessed with this and tried various things. I am not even 100% sure if I ended up fixing this fully (there are still weirdnesses like it keeps the list in the textbox of the Send tab UI even though it doesn't necessarily use those coins if they don't pass the filter).

I did however mess with the coin chooser since opening this and beefed up the filtering rules -- so that likely did it.  I'll close.
